### PR TITLE
[YSQL][DOCS] #2029 #2036 Document triggers and foreign keys

### PR DESCRIPTION
**Pull request by m-iancu**

### CODE REVIEW

File: docs/content/latest/api/ysql/_index.md [hunk 1]
```diff
@@ -41,12 +41,14 @@ DDL statements define the structures in a database, change their definitions, as
 | [`CREATE SEQUENCE`](commands/ddl_create_sequence) | Create a new sequence generator |
 | [`CREATE TABLE`](commands/ddl_create_table) | Create a new table |
 | [`CREATE TABLE AS`](commands/ddl_create_table_as) | Create a new table |
+| [`CREATE TRIGGER`](ddl_create_trigger) | Create a new trigger |
```
Comment (ndeodhar) [inline docs/content/latest/api/ysql/_index.md]:
```suggestion
| [`CREATE TRIGGER`](commands/ddl_create_trigger) | Create a new trigger |
```

File: docs/content/latest/api/ysql/_index.md [hunk 2]
```diff
@@ -41,12 +41,14 @@ DDL statements define the structures in a database, change their definitions, as
...
 | [`CREATE TYPE`](commands/ddl_create_type) | Create a new type |
 | [`CREATE VIEW`](commands/ddl_create_view) | Create a new view |
 | [`DROP DATABASE`](commands/ddl_drop_database) | Delete a database from the system |
 | [`DROP SEQUENCE`](commands/ddl_drop_sequence) | Delete a sequence generator |
 | [`DROP TABLE`](commands/ddl_drop_table) | Delete a table from a database |
 | [`DROP TYPE`](commands/ddl_drop_type) | Delete a user-defined type |
+| [`DROP TRIGGER`](ddl_drop_trigger) | Delete a trigger |
```
Comment (ndeodhar) [inline docs/content/latest/api/ysql/_index.md]:
```suggestion
| [`DROP TRIGGER`](commands/ddl_drop_trigger) | Delete a trigger |
```

File: docs/content/latest/api/ysql/commands/_index.md [hunk 3]
```diff
@@ -33,16 +33,18 @@ The following SQL statements are supported by the YugaByte Structured Query Lang
...
 | [`CREATE TABLE`](ddl_create_table) | Create a new table |
 | [`CREATE TABLE AS`](ddl_create_table_as) | Create a new table |
+| [`CREATE TRIGGER`](ddl_create_trigger) | Create a new trigger |
 | [`CREATE TYPE`](ddl_create_type) | Create a new type |
 | [`CREATE USER`](dcl_create_user) | Create a new user (role) |
 | [`CREATE VIEW`](ddl_create_view) | Create a new view |
 | [`DEALLOCATE`](perf_deallocate) | Deallocate a prepared statement |
 | [`DELETE`](dml_delete) | Delete rows from a table |
-| [`DO`](commands/cmd_doc) | Execute an anonymous code block |
+| [`DO`](commands/cmd_do) | Execute an anonymous code block |
```
Comment (ndeodhar) [inline docs/content/latest/api/ysql/commands/_index.md]:
```suggestion
| [`DO`](cmd_do) | Execute an anonymous code block |
```

File: docs/content/latest/api/ysql/commands/ddl_create_table.md [hunk 4]
```diff
@@ -116,12 +116,52 @@ postgres=# CREATE TABLE cars(id int PRIMARY KEY,
...
 ### Table with foreign key constraint
 
+Define two tables with a foreign keys constraint.
 ```sql
 postgres=# CREATE TABLE products(id int PRIMARY KEY,
                                  descr text);
 postgres=# CREATE TABLE orders(id int PRIMARY KEY,
                                pid int REFERENCES products(id) ON DELETE CASCADE,
                                amount int);
+
```
Comment (ndeodhar) [inline docs/content/latest/api/ysql/commands/ddl_create_table.md]: Can you also update the grammar and diagram files to include `references_clause`?

File: docs/content/latest/api/ysql/syntax_resources/ysql_grammar.ebnf [hunk 5]
```diff
@@ -158,6 +158,19 @@ create_table_as ::= 'CREATE' 'TABLE' [ 'IF' 'NOT' 'EXISTS' ] \
                            table_name [ '(' column_name { ',' column_name } ')' ] \
                            'AS' query [ 'WITH' [ 'NO' ] 'DATA' ] ;
 
+
+(* CREATE TRIGGER *)
+create_trigger = 'CREATE' 'TRIGGER' name
```
Comment (ndeodhar) [inline docs/content/latest/api/ysql/syntax_resources/ysql_grammar.ebnf]:
```suggestion
create_trigger = 'CREATE' [ 'CONSTRAINT' ] 'TRIGGER' name
```

File: docs/content/latest/api/ysql/syntax_resources/ysql_grammar.ebnf [hunk 6]
```diff
@@ -158,6 +158,19 @@ create_table_as ::= 'CREATE' 'TABLE' [ 'IF' 'NOT' 'EXISTS' ] \
...
+                  ( 'BEFORE' | 'AFTER' | 'INSTEAD' 'OF' ) ( event { 'OR' event } ) \
+                      'ON' table_name
+                      [ 'FROM' table_name ]
+                      [ 'NOT' 'DEFERRABLE' ]
+                      [ 'FOR' [ 'EACH' ] ( 'ROW' | 'STATEMENT' ) ] \
+                      [ 'WHEN' ( condition ) ]
```
Comment (ndeodhar) [inline docs/content/latest/api/ysql/syntax_resources/ysql_grammar.ebnf]:
```suggestion
                      [ 'WHEN' '(' condition ')' ]
```

File: docs/content/latest/api/ysql/syntax_resources/ysql_grammar.ebnf [hunk 7]
```diff
@@ -472,6 +492,8 @@ name ::= '<Text Literal>' ;
 
 column_names ::= column_name { ',' column_name } ;
 
+condition = expression ;
```
Comment (ndeodhar) [inline docs/content/latest/api/ysql/syntax_resources/ysql_grammar.ebnf]:
```suggestion
condition ::= expression ;
```

File: docs/content/latest/api/ysql/syntax_resources/ysql_grammar.ebnf (hunk 5 above)
Comment (m-iancu) [reply docs/content/latest/api/ysql/syntax_resources/ysql_grammar.ebnf]: Omitted intentionally as we don't support it yet.

File: docs/content/latest/api/ysql/syntax_resources/ysql_grammar.ebnf (hunk 7 above)
Comment (m-iancu) [reply docs/content/latest/api/ysql/syntax_resources/ysql_grammar.ebnf]: Both versions are correct, trying to migrate to just `=` as it is simpler.

**Pull request merged**